hazelcast/src/main/java/com/hazelcast/query/impl/getters/Extractors.java

```diff
@@ -75,11 +75,16 @@ public Object extract(Object target, String attributeName, Object metadata) {
         return extract(target, attributeName, metadata, true);
     }
 
-    public Object extract(Object target, String attributeName, Object metadata, boolean failOnMissingReflectiveAttribute) {
+    public Object extract(Object target, String attributeName, Object metadata, boolean failOnMissingReflectiveAttribute,
+                          boolean useLazyDeserialization) {
```

Comment:
can't we pass `useLazyDeserialization` as metadata here, instead of introducing a new parameter? I feel like metadata is put here for these kinds of situations

Reply:
Done.